### PR TITLE
fix data column size error when creating const column in decimal_round function

### LINKED ISSUE #5013: query crash
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS g10;
CREATE DATABASE g10;
USE g10;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NULL ,c_0_1 BOOLEAN NOT NULL ,c_0_2 DECIMAL(15, 2)  NOT NULL ,c_0_3 BIGINT NULL ,c_0_4 LARGEINT NOT NULL ,c_0_5 BOOLEAN NOT NULL ,c_0_6 LARGEINT NOT NULL ,c_0_7 BOOLEAN NULL ,c_0_8 BOOLEAN NOT NULL ,c_0_9 FLOAT NOT NULL , INDEX index1_c_0_1(`c_0_1`) USING BITMAP, INDEX index1_c_0_4(`c_0_4`) USING BITMAP) UNIQUE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7,c_0_8) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 91 day)) DISTRIBUTED BY HASH (c_0_5,c_0_3,c_0_8,c_0_0,c_0_7) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_0_4") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATETIME NOT NULL,c_1_1 TINYINT NOT NULL,c_1_2 LARGEINT NOT NULL,c_1_3 ARRAY<INT> NOT NULL,c_1_4 BOOLEAN NOT NULL,c_1_5 DECIMAL(37, 13)  NOT NULL,c_1_6 ARRAY<INT> NOT NULL,c_1_7 DATE NOT NULL,c_1_8 INT NOT NULL) PRIMARY KEY (c_1_0) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 156 day)) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 CHAR(1) NULL ,c_2_1 VARCHAR(1) NULL ,c_2_2 BOOLEAN NOT NULL ,c_2_3 DATETIME NULL ,c_2_4 VARCHAR(11) NULL ,c_2_5 DATE NULL ,c_2_6 SMALLINT NOT NULL ,c_2_7 DOUBLE NULL ,c_2_8 SMALLINT NOT NULL ,c_2_9 DECIMAL(32, 15)  NULL ,c_2_10 BOOLEAN NOT NULL ,c_2_11 DATETIME NULL ,c_2_12 ARRAY<INT> NULL ,c_2_13 DATETIME NULL , INDEX index1_c_2_1(`c_2_1`) USING BITMAP) UNIQUE KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4,c_2_5,c_2_6) PARTITION BY RANGE(c_2_3) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 101 day)) DISTRIBUTED BY HASH (c_2_0,c_2_5,c_2_1) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t2 VALUES ("", "", true, "2010-01-07 08:00:14", "x", "2010-01-30", 1, 0.7627156778320756, 3, NULL, true, NULL, [-936097645, -936097645], "2010-01-18 08:00:14"), ("", NULL, false, "2010-01-20 08:00:14", "", "2010-01-25", 14, NULL, 1, 6.847076441592526, false, "2010-01-15 08:00:14", [-1107963050], "2010-01-20 08:00:14");
INSERT INTO t2 VALUES (NULL, "", true, "2010-01-15 08:00:14", "", "2010-01-29", 12, 0.4642056889884044, 11, 75.526126332912087, false, "2010-01-20 08:00:14", [905085756, 905085756, 905085756, 943510688, -257792095, -936097645, -1729087015, -257792095, -83992377], "2010-01-02 08:00:14"), ("", NULL, true, "2010-01-14 08:00:14", "xjkBOU0V", NULL, 4, -2.57792095E8, 7, 33.370572144959057, true, "2010-01-23 08:00:14", [-1729087015], "2010-01-18 08:00:14");
INSERT INTO t1 VALUES ("2010-01-27 08:00:14", 0, -1107963050, [-1107963050, -1107963050, 1848800054, 24232647, 943510688, -83992377, -880311523, 1567553231, 1407222619, -664772504, -1157666515, 1838103623, 240649421], true, 92824851757.8222021103624, [0, 712539128, -1982755950, -514375388, 1678498327, -1882098518], "2010-01-09", 16);
TRUNCATE TABLE t1;
INSERT INTO t0 VALUES ("2010-01-19 08:00:14", false, 43856981759.93, 1759386659, -1945340967, false, 351802706, false, true, 0.4642056889884044);
INSERT INTO t1 VALUES ("2010-01-18 08:00:14", 2, 240649421, [1826797260, 712539128, -588578103, 1678498327, 1901262848, -86951723, -1719783600, 1407222619, 1848800054, -1945340967, -83992377, 352710099, -1574976215, 1373115453], false, 19928405874.5707426396712, [1926533188, -1737919957, -82315902, 64558173, 1678498327, -514375388, -202099670, -1838222995, -1449038021, -895000403, -601740965, -1321780413, 352710099], "2010-01-05", 29), ("2010-01-29 08:00:14", 3, 1999455607, [-1999111616, -1611599796, -1278748100, -1729087015, 536468689, 1358156089, 1759386659], true, 19099489335.8027439969845, [140931831, -983643911, 199254195, -1560907865, 1102464547, 943510688, -706977262, -914709705, -1047629440, -1982755950], "2010-01-13", 8);
INSERT INTO t2 VALUES ("", "", false, "2010-01-29 08:00:14", "v", "2010-01-20", 7, 0.5970035454759273, 1, 2.620198853715250, false, "2010-01-18 08:00:14", [-914709705, 1601733890, 1696667863, -1309222218, 1945645851, -603603655], "2010-01-26 08:00:14");
INSERT INTO t0 VALUES ("2010-01-16 08:00:14", true, 7222535075.79, 1601733890, -1, true, -1402696462, true, false, 1.678498327E9), ("2010-01-15 08:00:14", true, 38635888214.02, NULL, 610293604, true, -1882098518, true, true, 0.6103678002530841);
INSERT INTO t1 VALUES ("2009-12-31 08:00:14", 6, 1826797260, [472961466, 352710099, -2006578202, -1838222995, 1131432008, -685107177, -49294647, 1110910097, -309945173, 1157795468, -1872080293, -1601922913, -1838222995, -260588537], true, 50867173484.4224232574568, [1675171744], "2010-01-10", 0), ("2010-01-04 08:00:14", 1, 794069506, [-2075615723, 1999455607, 198431362, -983643911, -1560907865, 905085756, -337000604, -1999111616, 1131432008, -1309222218, -1320697469, -309945173, -514375388], true, 34412760925.1184907181509, [-2091017498, -1611599796, 872841541, 199254195, -1386647920], "2010-01-29", 18);
INSERT INTO t2 VALUES ("", "", true, NULL, "C", "2010-01-05", 5, 0.4642056889884044, 1, 25.103013864782798, false, "2010-01-22 08:00:14", [-9223372036854775808, 1352622237, 352710099, -473023310, -2075615723, 1476736112, 1446039463, 1276339487, -1817153796, 1901262848], "2010-01-04 08:00:14"), ("", "", false, NULL, "YH44dZd", "2010-01-17", 1, -2.006578202E9, 5, 51.814186430113414, false, "2010-01-13 08:00:14", [-12096350, 140931831, 2145746494, 1542467904, -1090813311, 1841802780, 236149530, -133164248, 1358156089, -1729087015, 1118229052, 1696667863, -708067417, 1074181637], "2010-01-13 08:00:14");
INSERT INTO t1 VALUES ("2010-01-20 08:00:14", 4, -1999111616, [943510688, -722911990, 1675171744, 1476736112, 1524715664, 1046324436, -1070108579, -1063870058, -577983635, 1131432008, 1901262848, -834613538, -1701237190, -533552076], true, 68919761213.9370363602751, [-1408751988, -257792095, 1826797260, -373628878], "2010-01-12", 0);
INSERT INTO t1 VALUES ("2010-01-10 08:00:14", 1, 64558173, [24232647, 1678498327, -791895094, -1008010870, 1044656602, -2043483234, 1700109646], false, 99951803033.1199014505545, [352710099, 92004420, 1224940164, -1557185428, 1362082625, -1839752511, 1106147358, 1641595210, -671769483, 16073683, -564995678, -983643911, -1942428024], "2010-01-13", 20);
INSERT INTO t1 VALUES ("2010-01-15 08:00:14", 2, 940254574, [1838499199, -1965848860, -2091017498, -1631840579, -692275710, -866493173, -1, 64558173, 1848800054, 477697526], false, 51606717967.0404385722194, [-514375388, -543926204, -1086731584], "2010-01-07", 2), ("2010-01-17 08:00:14", 1, 1567553231, [-1335762826, -697610027, 344951921, 304198151, 1567100146, -1082999435, 1883509335, -1205049419, -1761304813, -2129689421, 24232647, 341755129], true, 30156604854.0227171785215, [-1191949656, 269602238, -1634969251, -1817153796], "2010-01-08", 7), ("2010-01-23 08:00:14", 0, -1972822512, [2009508815, 1382721818, 794069506, -1808165973], false, 81768387992.6537571243379, [-12096350, 1926533188, -1107963050, -268963898, -462926360, -2010028154, 120965730, -603603655, 1601733890, -1645129425, -1402696462, -1745883994, -1982755950, 1826797260], "2010-01-02", 16);
INSERT INTO t0 VALUES ("2010-01-08 08:00:14", true, 87317062168.92, 2017262845, -444555630, true, 134512856, false, false, 0.5405943398289546);
INSERT INTO t0 VALUES ("2010-01-30 08:00:14", true, 29557409835.18, 1372541014, 837822980, true, 290831533, true, false, 0.7627156778320756), ("2010-01-12 08:00:14", true, 91710733740.00, 476762661, -72612832, true, 1102464547, false, true, -4.9294647E7);
INSERT INTO t1 VALUES ("2010-01-14 08:00:14", 7, 422843441, [88521836, -1187594444, -1], false, 14617907376.3118339004861, [140931831, -1545355366, 1131432008, 1401008901, 1633084131, -2003711665, 380729352, 610293604, 140931831, -1409713590, 1522061444, 1819008850, -1450607433], "2010-01-13", 16), ("2010-01-16 08:00:14", 4, 264766936, [-532221783, -825721165, 797433981, -952216012, -1999111616, 1131432008, 327253, -765132008, -133164248, 601672904, -403748626], true, 6589088662.8526601498436, [1110910097, -1997919913, -1823932055, 1868304180, 1926533188, -1390136274, -1288074978], "2010-01-11", 25), ("2010-01-24 08:00:14", 6, 538281522, [1627247110], true, 70850544085.4597878883059, [7337320, 1945645851, 1515207613, -1872080293, -1123509613, 0, -9223372036854775808, -12096350, -1611599796, -343313075], "2010-01-19", 26);
INSERT INTO t2 VALUES ("", "", true, "2009-12-31 08:00:14", "Foobx2JI", NULL, 6, 0.7656601713969599, 6, 94.676779814034219, true, NULL, [-1601922913, 813998378, 248689741, -1165595331, 787401129, 1580421611, 1009291864], "2010-01-09 08:00:14"), (NULL, "", true, NULL, "sGr", "2010-01-09", 10, NULL, 12, 93.785798621183908, false, "2009-12-31 08:00:14", [-111535319, 1749926961, -708067417, -82315902, 1131432008, -386609974, -603603655, -82315902, -1079322446, 948625982, 807381745, -202099670], "2010-01-16 08:00:14");
INSERT INTO t2 VALUES ("", "", false, "2010-01-25 08:00:14", "y6O", "2010-01-23", 8, 0.15006890432251818, 6, 13.619902912459600, false, "2010-01-26 08:00:14", [-83992377, -708067417, 163241888, 1373115453, -1945340967, -1719783600, 1325852321, -1588630079, -891025701, -1132563470], "2010-01-03 08:00:14"), ("", "", false, "2010-01-12 08:00:14", "dS6l", "2010-01-29", 13, 0.6678047482832165, 5, NULL, false, NULL, [-708067417, 1094031023, 1523418760, 1848800054, -82315902, 150033564], "2010-01-22 08:00:14"), ("", "", false, "2010-01-04 08:00:14", "h", NULL, 15, 0.4642056889884044, 10, 55.294075040719686, true, "2010-01-29 08:00:14", [198431362, 1403707458, 125545675, 567034198, 1281354597, -1601922913], "2010-01-20 08:00:14");
INSERT INTO t0 VALUES (NULL, false, 92825604337.04, -1574976215, -410672481, true, 1138732423, true, true, 0.7627156778320756), ("2010-01-08 08:00:14", false, 52371011312.49, -1107963050, -1297409948, false, -492359877, false, true, 0.3213476053818657);
INSERT INTO t1 VALUES ("2010-01-10 08:00:14", 0, 732615488, [1242505095, 1606315142, 1687520327], false, 80693457149.8377550830741, [-337000604, -1719783600, -1882098518, -603603655, 536468689, 1101356730, -911963700, 2145746494, -337000604], "2010-01-15", 14);
INSERT INTO t1 VALUES ("2010-01-06 08:00:14", 5, -1752578639, [-2140455404, -309945173, -534965637, 1446039463, -884816730], false, 95397179855.6477068665603, [2144191867, 534509482, -983643911, -2015360643, -621552399, 1589285478, 1378001918, -1107820845, 872841541, -908984964, -988879907, 1545373627, 536468689, -139613769], "2010-01-21", 3), ("2010-01-02 08:00:14", 3, 1194935243, [813027480, -248534839, -1913437415, 1678498327, -260588537, 1696667863, 1252345358, 1678498327, 1474720185, -65194911, -330604822, -2008787325, 1750732171], false, 18276604338.8927101888463, [-880311523, 176701692, 1056423576, -601740965], "2010-01-04", 7);
INSERT INTO t0 VALUES ("2010-01-22 08:00:14", false, 86219130733.14, 1221983766, 1476736112, true, 118141758, true, true, 0.6678047482832165), ("2010-01-12 08:00:14", true, 99726032862.57, 905085756, 198431362, true, -1644139395, true, false, 0.49447259397087784);
INSERT INTO t2 VALUES ("", NULL, true, "2010-01-18 08:00:14", "t7", "2010-01-12", 5, 0.9548766161482899, 9, 26.377307672508771, true, "2010-01-05 08:00:14", [-1685988568, 2033394569, 585010481, 1672489780, 1074181637, 359378028, 780539632, 190612531, -5471144, -254531418, -876508789], "2010-01-19 08:00:14"), ("", "", false, "2010-01-16 08:00:14", "S", NULL, 15, 0.47624409576949245, 2, 95.186550025568723, true, "2010-01-04 08:00:14", [1131432008, -1843371456, 1253525965, -202099670, 1118229052, -518166259, -83992377], "2010-01-26 08:00:14"), ("", "", false, "2010-01-17 08:00:14", "8", "2010-01-12", 3, 0.7227125764467922, 5, 0.571781556894657, false, "2010-01-22 08:00:14", [-1175375022, -663737917, -1047629440, 457064924, -1066951774, -63479164, -1125117342, -1583904312, -1107963050, -1020081558], "2010-01-08 08:00:14"), ("", "", false, "2010-01-16 08:00:14", "kXXS", "2010-01-17", 7, 0.38810319261570336, 7, 71.786382481408783, true, "2010-01-01 08:00:14", [-1817153796, -1226902384, 846391954], NULL);
 SELECT t2.c_2_2, t2.c_2_10, t2.c_2_13, t2.c_2_11, t2.c_2_5, t2.c_2_12, t2.c_2_3, t2.c_2_9, t2.c_2_4, t2.c_2_6, t2.c_2_0 FROM t2 WHERE (t2.c_2_11)  BETWEEN (MINUTES_ADD("1970-01-10 09:58:28", ROUND(0.3400760050182692) ) ) AND (t2.c_2_11)  GROUP BY t2.c_2_2, t2.c_2_10, t2.c_2_13, t2.c_2_11, t2.c_2_5, t2.c_2_12, t2.c_2_3, t2.c_2_9, t2.c_2_4, t2.c_2_6, t2.c_2_0 UNION SELECT t2.c_2_2, t2.c_2_10, t2.c_2_13, t2.c_2_11, t2.c_2_5, t2.c_2_12, t2.c_2_3, t2.c_2_9, t2.c_2_4, t2.c_2_6, t2.c_2_0 FROM t2 WHERE (NOT ((t2.c_2_11)  BETWEEN (MINUTES_ADD("1970-01-10 09:58:28", ROUND(0.3400760050182692) ) ) AND (t2.c_2_11) )) GROUP BY t2.c_2_2, t2.c_2_10, t2.c_2_13, t2.c_2_11, t2.c_2_5, t2.c_2_12, t2.c_2_3, t2.c_2_9, t2.c_2_4, t2.c_2_6, t2.c_2_0 UNION SELECT t2.c_2_2, t2.c_2_10, t2.c_2_13, t2.c_2_11, t2.c_2_5, t2.c_2_12, t2.c_2_3, t2.c_2_9, t2.c_2_4, t2.c_2_6, t2.c_2_0 FROM t2 WHERE (((t2.c_2_11)  BETWEEN (MINUTES_ADD("1970-01-10 09:58:28", ROUND(0.3400760050182692) ) ) AND (t2.c_2_11) ) IS NULL) GROUP BY t2.c_2_2, t2.c_2_10, t2.c_2_13, t2.c_2_11, t2.c_2_5, t2.c_2_12, t2.c_2_3, t2.c_2_9, t2.c_2_4, t2.c_2_6, t2.c_2_0;

```

```
*** Aborted at 1649817649 (unix time) try "date -d @1649817649" if you are using GNU date ***
PC: @          0x310e4f1 starrocks::vectorized::FunctionHelper::union_nullable_column()
*** SIGSEGV (@0x23) received by PID 1996 (TID 0x7fd79d3ac700) from PID 35; stack trace: ***
    @     0x7fd7c8cfd630 (unknown)
    @          0x310e4f1 starrocks::vectorized::FunctionHelper::union_nullable_column()
    @          0x344d89f starrocks::vectorized::TimeFunctions::minutes_add()
    @          0x310bb4c starrocks::vectorized::VectorizedFunctionCallExpr::evaluate()
    @          0x2c6fb04 starrocks::vectorized::VectorizedBinaryPredicate<>::evaluate()
    @          0x30c556e starrocks::vectorized::VectorizedOrCompoundPredicate::evaluate()
    @          0x19f8404 starrocks::ExprContext::evaluate()
    @          0x23c5ea0 starrocks::ExecNode::eval_conjuncts()
    @          0x282e3fb starrocks::pipeline::OlapChunkSource::_read_chunk_from_storage()
    @          0x282e79e starrocks::pipeline::OlapChunkSource::buffer_next_batch_chunks_blocking()
    @          0x26d1a97 _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS1_12RuntimeStateEiEUlvE0_E9_M_invokeERKSt9_Any_data
    @          0x1ed2a40 starrocks::PriorityThreadPool::work_thread()
    @          0x3d38187 thread_proxy
    @     0x7fd7c8cf5ea5 start_thread
    @     0x7fd7c83108dd __clone
    @                0x0 (unknown)

```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.2 Release
 - mysql> select current_version();
+---------------------------------+
| current_version()               |
+---------------------------------+
| SQLANCER_01_RELEASE 145a79c |
+---------------------------------+

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5013

## Problem Summary(Required) ：

This issue is caused when evaluating function `minutes_add('1970-01-10 09:58:28', CAST(round(0.3400760050182692) AS INT)))`.

The following is a step-by-step analysis of this issue.

1. `round` function will return a const column which only the value of 0th element is valid. 
2. `CAST(round(0.3400760050182692) AS INT)` will  call `DealNullableColumnUnaryFunction<UnpackConstColumnUnaryFunction<DecimalTo<true>>>` function.
But after executing `UnpackConstColumnUnaryFunction`, `DecimalTo<true>` will not perceive the existence of const and evaluate on all elements. If an overflow occurs, nullable column will be returned although it may appear on invalid elements. So the `UnpackConstColumnUnaryFunction` **may create a const nullable column for non-NULL constant** and return if data_column is nullable.

```cpp
template <typename FN>
class UnpackConstColumnUnaryFunction {
public:
    template <PrimitiveType Type, PrimitiveType ResultType, typename... Args>
    static inline ColumnPtr evaluate(const ColumnPtr& v1, Args&&... args) {
        if (v1->is_constant()) {
            auto eva1 = ColumnHelper::as_raw_column<ConstColumn>(v1)->data_column();
            ColumnPtr data_column = FN::template evaluate<Type, ResultType, Args...>(eva1, std::forward<Args>(args)...);

            return ConstColumn::create(data_column, v1->size());
        } else {
            return FN::template evaluate<Type, ResultType, Args...>(v1, std::forward<Args>(args)...);
        }
    }
};
```

3. Then `minutes_add` will call `UnionNullableColumnBinaryFunction<UnpackConstColumnBinaryFunction<OP>>` function. In `UnionNullableColumnBinaryFunction`, `FunctionHelper::get_data_column_of_nullable` will be called.
```cpp
    template <PrimitiveType LType, PrimitiveType RType, PrimitiveType ResultType>
    static ColumnPtr evaluate(const ColumnPtr& v1, const ColumnPtr& v2) {
        if (v1->only_null() || v2->only_null()) {
            return ColumnHelper::create_const_null_column(v1->size());
        }

        const ColumnPtr& data1 = FunctionHelper::get_data_column_of_nullable(v1);
        const ColumnPtr& data2 = FunctionHelper::get_data_column_of_nullable(v2);

        ColumnPtr data_result = FN::template evaluate<LType, RType, ResultType>(data1, data2);
        if (v1->has_null() || v2->has_null()) {
            // two NOT-nullable column can generate a {nullable, not-nullable, const, const-null} column
            // because when decimal overflow checking is enabled, overflow results will yield NULL.
            NullColumnPtr null_flags = FunctionHelper::union_nullable_column(v1, v2);
            return FunctionHelper::merge_column_and_null_column(std::move(data_result), std::move(null_flags));
        } else {
            return data_result;
        }
    }
```

4. In `FunctionHelper::get_data_column_of_nullable`, down_cast is performed on ptr.It is illegal to convert `ConstColumn` to `NullableColumn`. `v2`in step 3 is a const nullable column, this is unexpected, so coredump is happened.
```cpp
    static inline const ColumnPtr& get_data_column_of_nullable(const ColumnPtr& ptr) {
        if (ptr->is_nullable()) {
            return down_cast<NullableColumn*>(ptr.get())->data_column();
        }
        return ptr;
    }
``` 

After analysis, the root cause is that the const column returned by decimal_round function is invalid because its data_column size greater than 1. 
In this PR, I have fixed it and add check logic in ConstColumn's constructor

### CODE REVIEW

**Comment by kangkaisen** [comment]
if v2in step 3 is a const nullable column, the only_null check will be true. the code should run into 
```
  if (v1->only_null() || v2->only_null()) {
            return ColumnHelper::create_const_null_column(v1->size());
        }
```

**Comment by silverbullet233** [comment]
> if v2in step 3 is a const nullable column, the only_null check will be true. the code should run into
> 
> ```
>   if (v1->only_null() || v2->only_null()) {
>             return ColumnHelper::create_const_null_column(v1->size());
>         }
> ```

Yes, but this check is only valid for const nullable column with `NULL`. In this case, v2 is a const nullable column with value `0`.so `v2->only_null()` will return false.
Essentially, it is enough for non-null constant to be represented by const column, const nullable column should not be generated.

**Review by murphyatwork — DISMISSED** [review]

**Comment by kangkaisen** [comment]
```
v2 is a const nullable column with value 0
```
which isn't the valid const column. const column means literals in database, so for 0, the column should be const non-nullable int column.

**Comment by kangkaisen** [comment]
please add UT

**Comment by kangkaisen** [comment]
```
DealNullableColumnUnaryFunction<UnpackConstColumnUnaryFunction<DecimalTo<true>>>
```
if DecimalTo will result in nullable columns, seems it shouldn't use this template

**Review by kangkaisen — DISMISSED** [review]

File: be/src/column/const_column.cpp [hunk 1]
```diff
@@ -11,10 +11,12 @@
 namespace starrocks::vectorized {
 
 ConstColumn::ConstColumn(ColumnPtr data) : _data(std::move(data)), _size(0) {
+    DCHECK_LE(_data->size(), 1);
```
Comment (stdpain) [inline be/src/column/const_column.cpp:14]: remove this DCHECK

Comment (stdpain) [reply be/src/column/const_column.cpp:14]: BE may crash in some expression which create a column by ColumnBuilder

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
please remember cherry pick this bug fix to branch-2.2. @silverbullet233

**Review by murphyatwork — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by silverbullet233** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**